core/domain/classifier_services.py

```diff
@@ -162,58 +162,63 @@ def get_classifier_by_id(classifier_id):
     return classifier
 
 
-def _create_classifier(classifier):
+def create_classifier(job_id, classifier_data):
     """Creates classifier data model in the datastore given a classifier
        domain object.
 
     Args:
-        classifier: Domain object for the classifier.
+        job_id: str. ID of the ClassifierTrainingJob corresponding to the
+            classifier.
+        classifier_data: dict. The trained classifier data.
 
     Returns:
         classifier_id: str. ID of the classifier.
+
+    Raises:
+        Exception. The ClassifierDataModel corresponding to the job already
+            exists.
+        Exception. The algorithm_id of the job does not exist in the Interaction
+            Classifier Mapping.
     """
+    classifier_data_model = classifier_models.ClassifierDataModel.get(
+        job_id, strict=False)
+    if classifier_data_model is not None:
+        raise Exception(
+            'The ClassifierDataModel corresponding to the job already exists.')
+
+    classifier_training_job = get_classifier_training_job_by_id(job_id)
+    state_name = classifier_training_job.state_name
+    exp_id = classifier_training_job.exp_id
+    exp_version = classifier_training_job.exp_version
+    algorithm_id = classifier_training_job.algorithm_id
+    interaction_id = classifier_training_job.interaction_id
+    data_schema_version = None
+    if feconf.INTERACTION_CLASSIFIER_MAPPING[interaction_id][
+            'algorithm_id'] == algorithm_id:
+        data_schema_version = feconf.INTERACTION_CLASSIFIER_MAPPING[
+            interaction_id]['current_data_schema_version']
+    if data_schema_version is None:
+        raise Exception(
+            'The algorithm_id of the job does not exist in the Interaction'
```

Comment:
nit: missing space at end of this string. (You'll end up with "... InteractionClassifier Mapping".)